nixos/modules/security/wrappers/wrapper.c

```diff
@@ -151,9 +151,42 @@ static int make_caps_ambient(const char *self_path) {
     return 0;
 }
 
+// based on glibc sysdeps/generic/unsecvars.h
+const char **insecure_env = {
+    "GCONV_PATH",
+    "GETCONF_DIR",
+    "GLIBC_TUNABLES",
+    "HOSTALIASES",
+    "LD_AUDIT",
+    "LD_DEBUG",
+    "LD_DEBUG_OUTPUT",
+    "LD_DYNAMIC_WEAK",
+    "LD_HWCAP_MASK",
+    "LD_LIBRARY_PATH",
+    "LD_ORIGIN_PATH",
+    "LD_PRELOAD",
+    "LD_PROFILE",
+    "LD_SHOW_AUXV",
+    "LOCALDOMAIN",
+    "LOCPATH",
+    "MALLOC_TRACE",
+    "NIS_PATH",
+    "NLSPATH",
+    "RESOLV_HOST_CONF",
+    "RES_OPTIONS",
+    "TMPDIR",
+    "TZDIR",
+    NULL
+};
+
 int main(int argc, char **argv) {
     ASSERT(argc >= 1);
 
+    // Drop insecure environment variables
```

Comment:
I'd really prefer for this comment to explicitly state that musl (unlike glibc) does not remove these variables on its own, and that they must all be removed manually because they would introduce a massive hole. This is something that really must not be inadvertedly removed when cleaning up the panic fixes. It may also be a good time to adjusd the commit title.

Reply:
I think this comment suffices? I'm not sure what more should go in the title.